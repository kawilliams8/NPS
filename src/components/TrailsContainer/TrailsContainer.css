.TrailsContainer {
  background-color: #F5F5F5;
  border: .5px solid #989898;
  border-radius: 3px;
  font-size: 12px;
  height: 500px;
  margin: 10px 30px 10px 10px;
  overflow: scroll;
  padding: 10px;
  text-align: left;
  width: 45%;
}

.Trails-info-subheading {
  font-size: 18px;
}