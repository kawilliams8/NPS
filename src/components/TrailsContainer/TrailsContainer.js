import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Trail from '../Trail/Trail';
import './TrailsContainer.css';
import { fetchTrails } from '../../apiCalls/apiCalls';

export default class TrailsContainer extends Component {
  constructor(props){
    super(props)
    this.state = {
    lat: "",
    long: "",
    url: "",
    trails: []
  }
}

  componentDidMount () {
    let lat, long;
    if (this.props.data.latLong) {
        try {
        lat = this.props.data.latLong.split(" ")[0].substring(4, 12);
        long = this.props.data.latLong.split(" ")[1].substring(5, 15);
        this.setState({lat: lat, long: long})
      } finally {
        this.setState({ url: `https://www.trailrunproject.com/data/get-trails?lat=${lat}&lon=${long}&maxDistance=10&key=200590238-ef474523460511c3d882949ee2312c21` })
      }
    }
  }

  componentDidUpdate() {
    if (this.state.trails.length < 1) {
      fetchTrails(this.state.url)
      .then(data => this.setState({trails: data.trails}))
      .catch(error => this.setState({error: error.message}))
    }
  }
  
  render() {
    const trailCards = this.state.trails.map((trail, index) => {
      return <Trail trail={trail} key={index}/>
    });

    return (
      <section className="TrailsContainer">
        <h3 className="Trails-info-subheading">Hiking/Running Trails near {this.props.data.name}:</h3>
        {this.state.trails.length < 1 && <h4>No trails found near this National Park... Volunteer and build one!</h4>}
        {trailCards}
      </section>
    )
  }
};

TrailsContainer.propTypes = {
  props: PropTypes.shape({
    description: PropTypes.string,
    designation: PropTypes.string,
    directionsInfo: PropTypes.string,
    directionsUrl: PropTypes.string,
    fullName: PropTypes.string,
    id: PropTypes.string,
    latLong: PropTypes.string,
    name: PropTypes.string,
    parkCode: PropTypes.string,
    states: PropTypes.string,
    type: PropTypes.string,
    url: PropTypes.string,
    weatherInfo: PropTypes.string,
  })
}