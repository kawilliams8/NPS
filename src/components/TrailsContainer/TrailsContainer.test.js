import React from 'react';
import { shallow } from 'enzyme';
import TrailsContainer from '../TrailsContainer/TrailsContainer';

describe('TrailsContainer', () => {

  let wrapper, mockPark;
  beforeEach(() => {
    mockPark = {
      "states": "DC",
      "directionsInfo": "The memorial is located at the corner of...",
      "directionsUrl": "http://www.nps.gov/afam/planyourvisit/directions.htm",
      "url": "https://www.nps.gov/afam/index.htm",
      "weatherInfo": "Washington DC gets to see all four seasons...",
      "name": "African American Civil War Memorial",
      "latLong": "lat:38.916554, long:-77.025977",
      "description": "Over 200,000 African-American soldiers...",
      "designation": "",
      "parkCode": "afam",
      "id": "1A47416F-DAA3-4137-9F30-14AF86B4E547",
      "fullName": "African American Civil War Memorial",
      "type": "parks"
    }
    wrapper = shallow(<TrailsContainer 
      data={mockPark}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

})