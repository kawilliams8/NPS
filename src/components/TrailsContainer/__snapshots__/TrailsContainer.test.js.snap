// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrailsContainer should match the snapshot 1`] = `
<section
  className="TrailsContainer"
>
  <h3
    className="Trails-info-subheading"
  >
    Hiking/Running Trails near 
    African American Civil War Memorial
    :
  </h3>
  <h4>
    No trails found near this National Park... Volunteer and build one!
  </h4>
</section>
`;
